Update documentation (#417)
eclipse/lsp4jakarta: DEVELOPING.md

# Developing Eclipse LSP4Jakarta

- [Projects](#projects)
- [Prerequisites](#prerequisites)
- [Project setup in the Eclipse IDE](#project-setup-in-the-eclipse-ide)
- [Common errors](#common-errors)
- [Run and Debug](#run-and-debug)
- [Snippets](#snippets)

## Projects

This repository contains a few projects that provide language support for Jakarta EE APIs.

- [jakarta.ls](./jakarta.ls/)- Language Server for Jakarta EE
- [jakarta.jdt](./jakarta.jdt/) - [Eclipse JDT LS](https://github.com/eclipse/eclipse.jdt.ls/) extension for Eclipse LSP4Jakarta
- [jakarta.eclipse](./jakarta.eclipse) - Eclipse IDE client (for local testing) that consumes Eclipse LSP4Jakarta

To test the changes interactively, you must use a language client.

In the following sections, we document how to build and test using the Eclipse IDE client in this repository (./jakarta.eclipse).

## Prerequisites

> [JavaSE-17](https://adoptium.net/en-GB/marketplace/), [Maven](https://maven.apache.org/download.cgi), and [Eclipse](https://www.eclipse.org/downloads/) (Eclipse IDE for Enterprise Java and Web Developers is recommended) are required to build the Eclipse LSP4Jakarta project.

Reviewer: ```suggestion
[Java SE 17](https://adoptium.net/en-GB/marketplace/), [Maven](https://maven.apache.org/download.cgi), and [Eclipse](https://www.eclipse.org/downloads/) (Eclipse IDE for Enterprise Java and Web Developers is recommended) are required to build the Eclipse LSP4Jakarta project.
```

# Developing Eclipse LSP4Jakarta

- [Projects](#projects)
- [Prerequisites](#prerequisites)
- [Project setup in the Eclipse IDE](#project-setup-in-the-eclipse-ide)
- [Common errors](#common-errors)
- [Run and Debug](#run-and-debug)
- [Snippets](#contributing-snippets)

## Projects

This repo contains a few projects that provide language support for Jakarta EE APIs.

- [jakarta.ls](./jakarta.ls/)- Language Server for Jakarta EE
- [jakarta.jdt](./jakarta.jdt/) - Jakarta EE Eclipse [JDT LS](https://github.com/eclipse/eclipse.jdt.ls/) extension 
- [jakarta.eclipse](./jakarta.eclipse) - Eclipse IDE client (for local testing) that consumes the Language Server for Jakarta EE

To test the changes interactively, you must use a language client.

In the following sections, we document how to build and test using the Eclipse IDE client in this repository (jakarta.eclipse).

## Prerequisites

> [JavaSE-17](https://www.oracle.com/ca-en/java/technologies/downloads/#java17), [Maven](https://maven.apache.org/download.cgi), and [Eclipse](https://www.eclipse.org/downloads/) (Eclipse IDE for Enterprise Java and Web Developers is recommended) are required to build the Eclipse LSP4Jakarta project.

Reviewer: ```suggestion
[JavaSE-17](https://adoptium.net/en-GB/marketplace/), [Maven](https://maven.apache.org/download.cgi), and [Eclipse](https://www.eclipse.org/downloads/) (Eclipse IDE for Enterprise Java and Web Developers is recommended) are required to build the Eclipse LSP4Jakarta project.
```

Debug `org.eclipse.lsp4jakarta.lsp4e.core` as an Eclipse Application to launch a new instance of Eclipse with LSP4Jakarta (Right-click on the `org.eclipse.lsp4jakarta.lsp4e.core` project, "Debug As" --> "Eclipse Application"). A new Eclipse application will launch with the LSP4Jakarta Eclipse client plug-in installed.

<img src="/docs/images/building_debug_lsp4e.png" alt="Debug lsp4e Eclipse plug-in" height="60%" width="60%"/>

## Snippets

> Snippets are completion items that contain a block of helpful code for users. Snippets in Eclipse LSP4Jakarta are contributed through [JSON files](../jakarta.ls/src/main/resources/). If you add new snippets, update the [Jakarta EE API language features](../README.md#jakarta-ee-api-language-features) documentation

Reviewer 2: ```suggestion
Snippets are completion items that contain a block of helpful code for users. Snippets in Eclipse LSP4Jakarta are contributed through [JSON files](../jakarta.ls/src/main/resources/). If you add new snippets, update the [Jakarta EE API language features](../README.md#jakarta-ee-api-language-features) documentation.
```
